File: samples/java-sample-project/src/main/java/com/example/models/Notification.java
package com.example.models;

/**
 * A notification record (Java 16+).
 */
public record Notification(String id, String title, String message, NotificationType type) {

    /** Notification severity types. */
    public enum NotificationType {
        INFO,
        WARNING,
        ERROR
    }

    /** Creates an info notification. */
    public static Notification info(String id, String title, String message) {
        return new Notification(id, title, message, NotificationType.INFO);
    }
}

File: samples/typescript-sample-project/src/models/notification.ts
/** Notification severity types. */
export type NotificationType = "info" | "warning" | "error";

/** A notification message record. */
export interface Notification {
  readonly id: string;
  readonly title: string;
  readonly message: string;
  readonly type: NotificationType;
}

/** Creates an info notification. */
export function createNotification(
  id: string,
  title: string,
  message: string,
  type: NotificationType = "info"
): Notification {
  return { id, title, message, type };
}

/** Helper to create info notifications. */
export const createInfoNotification = (id: string, title: string, message: string): Notification =>
  createNotification(id, title, message, "info");
